pennylane/ops/qubit/matrix_ops.py

```diff
@@ -313,11 +352,39 @@ def compute_decomposition(D, wires):
 
         **Example:**
 
+        # TODO: Update once Identity has been replaced by GlobalPhase
         >>> qml.DiagonalQubitUnitary.compute_decomposition([1, 1], wires=0)
-        [QubitUnitary(array([[1, 0], [0, 1]]), wires=[0])]
+        [QubitUnitary(array([[0.36235775+0.93203909j, 0.        +0.j        ],
+                             [0.        +0.j        , 0.36235775+0.93203909j]]), wires=[0]),
+         RZ(1.5000000000000002, wires=[1]),
+         RZ(-0.10000000000000003, wires=[0]),
+         IsingZZ(0.2, wires=[0, 1])]
 
         """
-        return [QubitUnitary(DiagonalQubitUnitary.compute_matrix(D), wires=wires)]
+        n = len(wires)
+        phases = qml.math.real(qml.math.log(D) * (-1j))
+        coeffs = _walsh_hadamard_transform(phases, n).T
+        global_phase = qml.math.exp(1j * coeffs[0])
+        # For all other gates, there is a prefactor -1/2 to be compensated.
+        coeffs = coeffs * (-2.0)
+
+        ops = [QubitUnitary(qml.math.tensordot(global_phase, qml.math.eye(2), axes=0), wires[0])]
+        for wire0 in range(n):
+            # Single PauliZ generators correspond to the coeffs at powers of two
```

Comment:
Thanks for these comments I was quite confused about what was happening. Now I'm slightly less confused 🤣

Reply:
Haha, sorry about that...Also for future reference:
Consider the example `n=3`. Then the lexicographically ordered Pauli words consisting of `"Z"` and `"I"` are
```
["III", "IIZ", "IZI", "IZZ", "ZII", "ZIZ", "ZZI", "ZZZ"]
```
This means, we might as well write them as bit strings, indicating `"Z"` with a `1`:
```
[000, 001, 010, 011, 100, 101, 110, 111]
```
Now, the bit strings with a single `1` (single `"Z"`) are exactly powers of two, the ones with two `1`s are those that are a sum of two powers of two and so on :)

Reply:
The annoyance here is that we count up in the above order, but we typically reference wires in the opposite order:
`"ZII"` is a Z operator on the "first" qubit. Hence the mapping with `n-1-...` below.

Reply:
Thanks a lot for the explanation!